K2Bridge.Tests.UnitTests/Visitors/AggregationVisitorTests.cs

```diff
@@ -182,5 +182,33 @@ public string AggregationVisit_WithSubSumAgg_ReturnsSumAggregates()
 
             return aggregateClause.KustoQL;
         }
+
+        [TestCase(ExpectedResult = "['A%50.0']=percentile(fieldA, 50)")]
+        public string AggregationVisit_WithPercentileAgg_ReturnsPercentilePrimary()
+        {
+            var aggregateClause = new AggregationContainer()
+            {
+                PrimaryAggregation = new PercentileAggregation() { Field = "fieldA", Key = "A", Percents = new double[] { 50 }, Keyed = false },
+            };
+
+            var visitor = new ElasticSearchDSLVisitor(SchemaRetrieverMock.CreateMockSchemaRetriever());
+            visitor.Visit(aggregateClause);
+
+            return aggregateClause.KustoQL;
+        }
+
+        [TestCase(ExpectedResult = "['A%25.0']=percentile(fieldA, 25),['A%50.0']=percentile(fieldA, 50),['A%99.0']=percentile(fieldA, 99)")]
```

Comment:
As mentioned by @tamirkamara , It will be more efficient if you will use PercentileArray instead of multiple percentile operators.

I saw in the metric agg visitor that you are using the percentile array.
so I am not sure why its not reflected in this test.

Reply:
done